openwisp_monitoring/monitoring/tests/test_graphs.py

```diff
@@ -301,3 +309,48 @@ def test_bad_json_query_returns_none(self):
         m.write('00:14:5c:00:00:00')
         g.save()
         self.assertIsNone(g.json(time=1))
+
+    def test_register_invalid_chart_configuration(self):
+        with self.subTest('Registering with incomplete chart configuration.'):
+            with self.assertRaises(AssertionError):
+                register_chart('test_type', dict())
+
+        with self.subTest('Registering already registered chart configuration.'):
+            histogram = test_charts['histogram']
+            register_chart('histogram_test', histogram)
+            with self.assertRaises(ImproperlyConfigured):
+                register_chart('histogram_test', histogram)
+            unregister_chart('histogram_test')
+
+        with self.subTest('Registering with improper chart configuration name'):
+            with self.assertRaises(ImproperlyConfigured):
+                register_chart(['test_type'], dict())
+
+        with self.subTest('Registering with improper chart configuration'):
+            with self.assertRaises(ImproperlyConfigured):
+                register_chart('test_type', tuple())
+
+        with self.subTest('Unregistering with improper chart configuration name'):
+            with self.assertRaises(ImproperlyConfigured):
+                unregister_chart(dict())
+
+        with self.subTest('Unregistering unregistered chart configuration'):
+            with self.assertRaises(ImproperlyConfigured):
+                unregister_chart('test_chart')
+
+    def test_register_valid_chart_configuration(self):
+        dummy = test_charts['dummy']
+        register_chart('dummy_test', dummy)
+        self.assertEqual(CHART_CONFIGURATION_CHOICES[-1], ('dummy_test', 'dummy_test'))
+        unregister_chart('dummy_test')
```

Comment:
Can you check after unregistering that the dummy test is not part of the CHOICES anymore?